artio-codecs/src/main/java/uk/co/real_logic/artio/util/MutableAsciiBuffer.java

```diff
@@ -45,6 +47,30 @@
         999999999_999999999L, Long.MAX_VALUE
     };
 
+    private static final long[] CEIL_VAL_AT_SCALE =
+    {
+        0L, // scale=0 , irrelevant entry, just place holder
+        10L,
+        100L,
+        1_000L, // scale=3
+        10_000L,
+        100_000L,
+        1_000_000L,  // scale=6
+        10_000_000L,
+        100_000_000L,
+        1_000_000_000L,  // scale=9
+        10_000_000_000L,
+        100_000_000_000L,
+        1_000_000_000_000L, // scale=12
+        10_000_000_000_000L,
+        100_000_000_000_000L,
+        1_000_000_000_000_000L, // scale=15
+        10_000_000_000_000_000L,
+        100_000_000_000_000_000L,
+        1_000_000_000_000_000_000L, // scale=18
+        //10_000_000_000_000_000_000L, // scale=19
```

Comment:
Why is this commented out?

Reply:
cannot really represent the value of 10 * 10^18.
look at method **checkForLeadingZeroAndPutIfNeeded** and see that if scale is GE 19 there is special treatment.

Reply:
In that case it's probably better to just remove the scale=19 entry.

Reply:
removed entirely this block.

Reply:
I removed the entire block, with simpler implementation.